### PR TITLE
Emit code in ESM format

### PR DESCRIPTION
**Pull request by vkrol**
Right now, the library is not tree shakeable. These changes will enable tree shaking.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "compilerOptions": {
     "module": "commonjs",
+    "moduleResolution": "node",
```
Comment (vkrol) [inline tsconfig.json]: It is needed for compiling to ESM.

**Comment by xnimorz** [comment]
Great work!
Thank you, @vkrol !
I'll publish it with your previous PR https://github.com/xnimorz/use-debounce/pull/33

**Comment by vkrol** [comment]
You are welcome. I can rebase this branch when https://github.com/xnimorz/use-debounce/pull/33 will be merged.

**Comment by xnimorz** [comment]
I've published `use-debounce@3.0.0` with these changes. 

Thank you for contributing!

**Comment by vkrol** [comment]
Awesome. No problem!

**Pull request merged**